packages/core/src/metadata/directives.ts

```diff
@@ -787,8 +793,8 @@ export interface InputDecorator {
    *
    * @see [Input and Output properties](guide/inputs-outputs)
    */
-  (bindingPropertyName?: string): any;
-  new(bindingPropertyName?: string): any;
+  (arg?: string|{publicName?: string, required?: boolean}): any;
```

Comment:
We could use the `Input` interface itself as the type here:
```suggestion
  (arg?: string|Input): any;
```